modules/vaos/app/services/vaos/v2/appointments_service.rb

```diff
@@ -131,6 +139,111 @@ def mobile_facility_service
           VAOS::V2::MobileFacilityService.new(user)
       end
 
+      def avs_service
+        @avs_service ||=
+          Avs::V0::AvsService.new
+      end
+
+      # Extracts the station number and appointment IEN from an Appointment.
+      #
+      # Given an appointment, this method will check the identifiers, find the identifier associated
+      # with 'VistADefinedTerms/409_84' and return the identifier value as a two-item array
+      # (split on the ':' character). If there is no such identifier, it will return nil.
+      #
+      # @param [Hash] appointment The appointment object to find the identifier in.
+      # This Hash must include an :identifier key.
+      #
+      # @return [Array, nil] An array containing two strings representing the station number
+      # and IEN if found, or nil if not.
+      def extract_station_and_ien(appointment)
+        return nil if appointment[:identifier].nil?
+
+        identifier = appointment[:identifier].find { |id| id[:system]&.include? 'VistADefinedTerms/409_84' }
+
+        return identifier[:value]&.split(':', 2) unless identifier.nil?
```

Comment:
Would this also make sense as 
```
return if identifier.nil?

identifier[:value]&.split(':', 2)
```

I think that would be a slightly clearer guard pattern.

Reply:
Yeah, whether you `return` or `return nil` doesn't matter to me. I just find the logic clearer when you guard on nil rather than return nil at the end.